src/t8_cmesh/t8_cmesh_copy.c

```diff
@@ -54,7 +54,7 @@ t8_cmesh_copy (t8_cmesh_t cmesh, const t8_cmesh_t cmesh_from, sc_MPI_Comm comm)
   cmesh->num_trees = cmesh_from->num_trees;
   cmesh->set_partition = cmesh_from->set_partition;
   cmesh->set_partition_level = cmesh_from->set_partition_level;
-  T8_ASSERT (t8_cmesh_comm_is_valid (cmesh, comm));
+  T8_ASSERT (t8_cmesh_get_mpicomm (cmesh) == comm);
```

Comment:
I would still use `t8_cmesh_comm_is_valid ` here because of the straighforward naming
```suggestion
  T8_ASSERT (t8_cmesh_comm_is_valid (cmesh, comm);
```